src/survey-public/utils/createSurveySchema.ts

```diff
@@ -0,0 +1,193 @@
+          }
+        } else {
+          schemaObject[`text-${id}`] = defaultZodSchema.text
+        }
+        break
+      case "readOnly": // input type text but read only
+        schemaObject[`text-${id}`] = validation?.optional
+          ? defaultZodSchema.text.nullish().or(z.literal(""))
+          : defaultZodSchema.text
+        break
+      case "text": // input type textarea
+        if (validation) {
+          let customTextareaZodSchmema = z
+            .string()
+            .trim()
+            .max(validation.maxLength || 10000, {
+              message: `Die maximale Zeichenanzahl von ${validation.maxLength || 10000} Zeichen ist überschritten. Bitte kürzen Sie Ihren Text entsprechend, um fortzufahren.`,
+            })
+          if (validation.minLength) {
+            customTextareaZodSchmema = customTextareaZodSchmema.min(validation.minLength, {
+              message: `Mindestens ${validation.minLength} Zeichen.`,
+            })
+          }
+          if (!validation.minLength && !validation?.optional) {
+            customTextareaZodSchmema = customTextareaZodSchmema.min(1, {
+              message: `Pflichtfeld.`,
+            })
+          }
+          if (validation?.optional) {
+            // @ts-expect-error this works // to do validtion ?
+            customTextareaZodSchmema = customTextareaZodSchmema.nullish().or(z.literal(""))
+          }
+          schemaObject[`text-${id}`] = customTextareaZodSchmema
+        } else {
+          schemaObject[`text-${id}`] = defaultZodSchema.textarea
+        }
+        break
+      case "singleResponse":
+        schemaObject[`single-${id}`] = validation?.optional
+          ? z.string({ message: "Pflichtfeld." }).trim().nullish().or(z.literal(""))
+          : defaultZodSchema.single
+        break
+      case "multipleResponse":
+        const multiResponseProps = props as TSingleOrMultiResponseProps
+        multiResponseProps.responses.forEach((r) => {
+          const key = `multi-${id}-${r.id}`
+          schemaObject[key] = z.boolean()
+          // Add key to multipleResponseKeys if not optional
+          // if it is optional we do not need to validate if one of the options is selected
+          // if (!validation?.optional) multipleResponseKeys.push(key)
+        })
+
+        break
+      case "custom":
+        schemaObject[`custom-${id}`] = z.union(
+          [
+            z.string().trim().min(1, { message: "Pflichtfeld." }),
+            z.number(),
+            z.boolean(),
+            z.object({}).passthrough(),
+            z.array(z.any()),
+          ],
+          { required_error: "Pflichtfeld." },
+        )
+        break
+      case "map":
+        schemaObject[`map-${id}`] = z.object(
+          {
+            lng: z.number(),
+            lat: z.number(),
+          },
+          { required_error: "Pflichtfeld." },
+        )
+        break
+    }
+  })
+
+  return z.object(schemaObject)
+
+  // if (!multipleResponseKeys?.length)
```

Comment:
Bitte LInk auf Issue in dem man mehr erfährt was wir mit diesem Teil machen. Oder den Teil löschen und in den Issue pasten.